t/lib/ljtestlib.pl

```diff
@@ -0,0 +1,92 @@
+package LJ;
+
+use v5.10;
+use strict;
+no warnings 'uninitialized';
+
+BEGIN {
+    # ugly hack to shutup dependent libraries which sometimes want to bring in
+    # ljlib.pl (via require, ick!).  so this lets them know if it's recursive.
+    # we REALLY need to move the rest of this crap to .pm files.
+
+    # ensure we have $LJ::HOME, or complain very vigorously
+    $LJ::HOME ||= $ENV{LJHOME};
+    die "No \$LJ::HOME set, or not a directory!\n"
+        unless $LJ::HOME && -d $LJ::HOME;
+
+    # Please do not change this to "LJ::Directories"
+    require $LJ::HOME . "/cgi-bin/LJ/Directories.pm";
+}
+
+use Test::MockObject;
+my $mock = Test::MockObject->new();
+
+# Fake some context functions from ljlib.pl
+
+sub is_enabled {
+    return 1;
+}
+
+sub is_web_context {
+    return 0;
+}
+
+# Fake user object support
+
+sub load_user_or_identity {
+    return undef;
+}
+
+sub canonical_username {
+    my ($user) = @_;
+    return $user;
+}
+
+sub ljuser {
+    my ( $user, $opts ) = @_;
+    return
+          "<span style='white-space: nowrap;'>"
+        . "<a href='http://lj.example/userprofile'><img src='http://lj.example/img.png' alt='[alttext]' width='60' height='80'"
+        . " style='vertical-align: text-bottom; border: 0; padding-right: 1px;' /></a>"
+        . "<a href='http://lj.example/url'>$user</a></span>";
+}
+
+sub temp_user {
```

Comment:
For this particular function, since you're not overriding a LJ::temp_user function in the main code, but rather providing an alternative to LJ::Test::temp_user, I think it would be less confusing if you put it in a different package like LJ::Mock instead of the main LJ package.

That's my only other suggestion for improvement - great work!

Reply:
done. please take a look - i couldn't do it other than by splitting into a new file and then adding the directory to `@INC`, not sure if that's the correct way